### PR TITLE
feat(cloudsql): Add SQL Server to Cloud SQL Cloud Run quickstart.

### PR DESCRIPTION
**Pull request by jsimonweb**

### CODE REVIEW

File: cloudsql/sqlserver/database-sql/Dockerfile [hunk 1]
```diff
@@ -43,6 +43,12 @@ RUN set -x && apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -
 # Copy the binary to the production image from the builder stage.
 COPY --from=builder /app/server /app/server
 COPY ./templates /app/templates
+# Copy any certificates if present.
+COPY ./certs /app/certs
+
+# Use server certificate for encrypted connection.
+COPY ./certs/server-ca.pem /usr/local/share/ca-certificates/server-ca.crt
+RUN update-ca-certificates
```
Comment (enocom) [inline cloudsql/sqlserver/database-sql/Dockerfile:51]: Is this step necessary?

File: cloudsql/sqlserver/database-sql/cloudsql.go [hunk 2]
```diff
@@ -238,6 +239,24 @@ func initTCPConnectionPool() (*sql.DB, error) {
...
+	// [START cloud_sql_sqlserver_databasesql_sslcerts]
+	// (OPTIONAL) Configure SSL certificates
+	// For deployments that connect directly to a Cloud SQL instance without
+	// using the Cloud SQL Proxy, configuring SSL certificates will ensure the
+	// connection is encrypted. This step is entirely OPTIONAL.
+	dbRootCert := os.Getenv("DB_ROOT_CERT") // e.g., '/path/to/my/server-ca.pem'
+	if dbRootCert != "" {
+		// Get connection host name to be matched with host name in SSL certificate.
+		var instanceConnectionName = mustGetenv("INSTANCE_CONNECTION_NAME")
+		hostNameParts := strings.Split(instanceConnectionName, ":")
```
Comment (enocom) [inline cloudsql/sqlserver/database-sql/cloudsql.go:252]: Should there be an error check to ensure there are three parts here?

**Review by enocom — APPROVED** [review]

File: cloudsql/sqlserver/database-sql/Dockerfile (hunk 1 above)
Comment (jsimonweb) [reply cloudsql/sqlserver/database-sql/Dockerfile:51]: Yes. This is how the server certificate is mounted within the container and used to match the certificate provided by the sample app when establishing a connection. Without this command, the sample app fails to connect with the error: "...TLS Handshake failed: x509: certificate signed by unknown authority".

File: cloudsql/sqlserver/database-sql/cloudsql.go (hunk 2 above)
Comment (jsimonweb) [reply cloudsql/sqlserver/database-sql/cloudsql.go:253]: Updated to add this.

File: cloudsql/sqlserver/database-sql/cloudsql.go [hunk 3]
```diff
@@ -238,6 +239,28 @@ func initTCPConnectionPool() (*sql.DB, error) {
...
+	// using the Cloud SQL Proxy, configuring SSL certificates will ensure the
+	// connection is encrypted. This step is entirely OPTIONAL.
+	dbRootCert := os.Getenv("DB_ROOT_CERT") // e.g., '/path/to/my/server-ca.pem'
+	if dbRootCert != "" {
+		// Get connection host name to be matched with host name in SSL certificate.
+		var instanceConnectionName = mustGetenv("INSTANCE_CONNECTION_NAME")
+		// Expected format of INSTANCE_CONNECTION_NAME is project_id:region:instance_name
+		hostNameParts := strings.Split(instanceConnectionName, ":")
+		if len(hostNameParts) != 3 {
+			log.Fatalf("Invalid format for INSTANCE_CONNECTION_NAME environment variable.")
```
Comment (enocom) [inline cloudsql/sqlserver/database-sql/cloudsql.go:255]: nit: adding what the user passed in might be nice:

``` go
log.Fatalf("Invalid format for INSTANCE_CONNECTION_NAME environment variable, got = %q", instanceConnectionName)
```

**Review by enocom — APPROVED** [review]

File: cloudsql/sqlserver/database-sql/cloudsql.go (hunk 3 above)
Comment (jsimonweb) [reply cloudsql/sqlserver/database-sql/cloudsql.go]: Done.

**Pull request merged**